jmclient/jmclient/taker_utils.py

```diff
@@ -97,38 +106,32 @@ def direct_send(wallet_service, amount, mixdepth, destination, answeryes=False,
             return
         total_inputs_val = sum([va['value'] for u, va in utxos.items()])
-
-        if is_burn_destination(destination):
-            if len(utxos) > 1:
-                log.error("Only one input allowed when burning coins, to keep "
-                    + "the tx small. Tip: use the coin control feature to freeze utxos")
-                return
-            address_type = FidelityBondMixin.BIP32_BURN_ID
-            index = wallet_service.wallet.get_next_unused_index(mixdepth, address_type)
-            path = wallet_service.wallet.get_path(mixdepth, address_type, index)
-            privkey, engine = wallet_service.wallet._get_key_from_path(path)
-            pubkey = engine.privkey_to_pubkey(privkey)
-            pubkeyhash = Hash160(pubkey)
-
-            #size of burn output is slightly different from regular outputs
-            burn_script = mk_burn_script(pubkeyhash)
-            fee_est = estimate_tx_fee(len(utxos), 0, txtype=txtype, extra_bytes=len(burn_script)/2)
-
-            outs = [{"script": burn_script, "value": total_inputs_val - fee_est}]
-            destination = "BURNER OUTPUT embedding pubkey at " \
-                + wallet_service.wallet.get_path_repr(path) \
-                + "\n\nWARNING: This transaction if broadcasted will PERMANENTLY DESTROY your bitcoins\n"
-        else:
-            #regular sweep (non-burn)
-            fee_est = estimate_tx_fee(len(utxos), 1, txtype=txtype, outtype=outtype)
-            outs = [{"address": destination, "value": total_inputs_val - fee_est}]
+        script_types = get_utxo_scripts(wallet_service.wallet, utxos)
+        fee_est = estimate_tx_fee(len(utxos), 1, txtype=script_types, outtype=outtype)
+        outs = [{"address": destination, "value": total_inputs_val - fee_est}]
     else:
-        #not doing a sweep; we will have change
-        #8 inputs to be conservative
-        initial_fee_est = estimate_tx_fee(8,2, txtype=txtype, outtype=outtype)
-        utxos = wallet_service.select_utxos(mixdepth, amount + initial_fee_est)
+        change_type = txtype
+        if custom_change_addr:
+            change_type = wallet_service.get_outtype(custom_change_addr)
+            if change_type is None:
+                # we don't recognize this type; best we can do is revert to default,
+                # even though it may be inaccurate:
+                change_type = txtype
+        if outtype is None:
+            # we don't recognize the destination script type,
+            # so set it as the same as the change (which will usually
+            # be the same as the spending wallet, but see above for custom)
+            outtype = change_type
+        outtypes = [change_type, outtype]
+        # not doing a sweep; we will have change.
+        # 8 inputs to be conservative; note we cannot account for the possibility
+        # of non-standard input types at this point.
+        initial_fee_est = estimate_tx_fee(8,2, txtype=txtype, outtype=outtypes)
```

Comment:
nit
```suggestion
        initial_fee_est = estimate_tx_fee(8, 2, txtype=txtype, outtype=outtypes)
```

Reply:
Done in https://github.com/JoinMarket-Org/joinmarket-clientserver/pull/1421/commits/357b61165551ac1e096527bc9f372c2c03377404